src/lib/permissions.ts

```diff
@@ -87,22 +59,16 @@ export function decodePermissions(permissionHex: string) {
   };
 
   const permissionsToTest = Object.keys(LSP6_DEFAULT_PERMISSIONS);
-  if (permissionHex === LSP6_ALL_PERMISSIONS) {
-    // Do not use an for of loop here to not require the regenerator runtime
-    for (let i = 0; i < permissionsToTest.length; i += 1) {
-      const testPermission = permissionsToTest[i];
-      result[testPermission] = true;
-    }
-    return result;
-  }
-
-  const passedPermissionDecimal = Number(hexToNumber(permissionHex));
+  const passedPermissionDecimal = BigInt(hexToNumber(permissionHex, true));
 
   // Do not use an for of loop here to not require the regenerator runtime
+  // Deal with ALL_PERMISSIONS the same way. So as long as all the bits in ALL_PERMISSIONS
```

Comment:
I am very confused by this comment. I think might need a bit of re-writing and maybe a small example?